dali/pipeline/data/tensor_vector.h

```diff
@@ -160,6 +220,16 @@ class DLL_PUBLIC TensorVector {
    */
   void SetSize(int new_size);
 
+  /**
+   * @brief Setup all the batch properties of this TensorVector the same way as the provided tensor:
+   *
+   * Precondition: the TensorVector should not have data.
+   * Configures: type, layout, pinned, order and dimensionality.
+   */
+  void SetupLike(const Tensor<Backend> &sample);
+
+  void SetupLike(const TensorVector<Backend> &other);
```

Comment:
Why isn't it a template? The function body is the same (or almost so). How about TensorList? Do we really need the argument to be on the same backend?

Reply:
This is mainly used for the purpose of preparing the object explicitly for `SetSample` - you won't `SetSample` across backends, using the same pinned configuration across backends when copying also doesn't make sense.

Reply:
OK with the backend - how about code duplication? I'd suggest this:

```C++
  void SetupLike(const Tensor<Backend> &sample) {
    SetupLikeImpl(sample);
  }
  void SetupLike(const TensorVector<Backend> &batch) {
    SetupLikeImpl(sample);
  }
  void SetupLike(const TensorList<Backend> &batch) {
    SetupLikeImpl(sample);
  }

 private:
  template <typename T>
  void SetupLikeImpl(const T &src) { do stuff }
```

Reply:
I can look into that. Will this be all?